bears/jinja2/Jinja2Bear.py

```diff
@@ -0,0 +1,271 @@
+
+
+def is_spacing_valid(string, required_spacing):
+    '''
+    Returns True iff `string` has exactly <required_spacing>
+    spaces at the start and at the end
+    '''
+    leading_spaces = len(string) - len(string.lstrip())
+    trailing_spaces = len(string) - len(string.rstrip())
+
+    return (
+        leading_spaces == required_spacing
+        and trailing_spaces == required_spacing)
+
+
+class Jinja2Bear(LocalBear):
+    LANGUAGES = {"Jinja2"}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Syntax'}
+    CAN_FIX = {'Formatting', 'Documentation'}
+
+    def handle_control_spacing_issue(self, file, filename, line, line_number,
+                                     control_spacing, match_object):
+        '''
+        Handles incorrectly spaced control tags (if and for),
+        both start and end tags
+
+        :param control_spacing: The number of spaces placed around control tags
+                                as required by the settings.
+        :param match_object:    the match object generated by a match on the
+                                regular expression control_start_regex or
+                                control_end_regex in the run method.
+        '''
+        diff = generate_spacing_diff(
+            file, filename, line, line_number, match_object, control_spacing)
+        return Result.from_values(
+            origin=self,
+            message="Control blocks should be spaced with "
+                    "`{0}` spaces on each side.".format(
+                        control_spacing),
+            file=filename,
+            line=line_number,
+            column=match_object.start(1) + 1,
+            end_line=line_number + 1,
+            end_column=match_object.end(1) + 1,
+            diffs=diff)
+
+    def run(self,
+            filename,
+            file,
+            variable_spacing: int=1,
+            control_spacing: int=1):
+        '''
+        Check Jinja2 templates for syntax, formatting and documentation issues.
+        The following aspects are being looked at:
+
+        * Variable spacing: Variable tags should be padded with one space on
+                            each side, like this: `{{ var_name }}`. This can
+                            be set to any number of spaces via the setting
+                            variable_spacing.
+                            Malformatted variable tags are detected and fixes
+                            suggested.
+        * Control spacing:  Like variable spacing, but for control blocks,
+                            i.e. if and for constructs. Looks at both start
+                            and end block.
+        * Control labels:   It is good practise to label the end of an if or
+                            for construct with a comment equal to the content
+                            of the start, like so:
+
+                            ```
```

Comment:
I think in RST This is done using `::` and indents.
Could you make the documentation locally and check if formatting is fine ?